### ISSUE TITLE
Bounding box in higher dimensions

### ISSUE THREAD

**Issue by rmcaixeta** [JuliaGeometry/Meshes.jl#1148]
Hello there, it's been long time!

I was testing some sequential gaussian simulations in higher dimensions (>=4). But it throws an error at the preparation/rescaling part because it can't create a bounding box in higher dimensions.

I tried replacing the functions _pboxes(::Type{𝔼{1}}, points), _pboxes(::Type{𝔼{2}}, points) and _pboxes(::Type{𝔼{3}}, points)
at https://github.com/JuliaGeometry/Meshes.jl/blob/master/src/boundingboxes.jl
for the following one

```julia
function _pboxes(::Type{𝔼{N}}, points) where {N}
  p = first(points)
  ℒ = lentype(p)
  cmin = fill(typemax(ℒ), N)
  cmax = fill(typemin(ℒ), N)

  for p in points
    c = getfield(convert(Cartesian, coords(p)), :coords)
    for i = 1:N
      cmin[i] = min(c[i], cmin[i])
      cmax[i] = max(c[i], cmax[i])
    end
  end
  Box(withcrs(p, Tuple(cmin)), withcrs(p, Tuple(cmax)))
end
```

and now it seems to work fine. It passes the tests, except for a few because that return an allocation a little bit higher
Can I send a pull request with that or do you see some problem with that approach?

**Comment by juliohm** [comment]
Hi @rmcaixeta , please feel free to submit a PR for review.

Notice that we only support CRS up to 3 spatial dimensions in CoordRefSystems.jl. It is an accident that our Gaussian simulation algorithm also works in N dimensions.

If you can think of ways to avoid memory allocation produced by `fill`, that would be even better. Performance tests with BenchmarkTools.jl are welcome. We can review the PR during the week.

**Comment by juliohm** [comment]
Closed as discussed in #1149
